site/en/blog/crx-scripting-api/index.md

```diff
@@ -0,0 +1,289 @@
+basically the final statement evaluated in a script. Think of it like the value returned when you
+execute a block of code in the Chrome DevTools console.
+
+In Manifest V2, `executeScript` and `insertCSS` would return an array of bare execution results.
+This is fine for a single injection point, but if you're injecting into all frames in a tab it can
+be difficult to track down which frame encountered the error.
+
+For a concrete example, let's take a look at the results returned by a Manifest V2 and Manifest V3.
+Both versions of the extension will share the same content script behavior. And for both examples,
+we'll be injecting on the same [demo page][iframe-demo].
+
+```js
+//// content-script.js
+var url = new URL(window.location.href);
+if (url.pathname.endsWith('.html')) {
+  throw new Error('Incompatible URL path!');
+}
+'success';
+```
+
+When we run the Manifest V2 version, we get back an array of `['success', null]`. Which result
+corresponds to the main frame and which is for the iframe? The return value doesn't tell us, so we
+don't know for sure.
+
+```js
+// Manifest V2 extension
+chrome.browserAction.onClicked.addListener((tab) => {
+  chrome.scripting.executeScript({
+    allFrames: true,
+    file: 'content-script.js';,
+  }, (results) => {
+    // results == ['success', null]
+    for (let result of results) {
+      if (result !== 'success') {
+        console.log('An error occurred, but where?');
+      }
+    }
+  });
+});
+```
+
+In the Manifest V3 version, results now contains an array of objects instead of an array of strings,
+and the result objects clearly identify the ID of the frame for each result. This makes it much
+easier for developers to identify and react to situations where injection fails unexpectedly.
+
+```js
+// Manifest V3 extension
+chrome.action.onClicked.addListener(async (tab) => {
+  let results = await chrome.scripting.executeScript({
+    target: {tabId: tab.id, allFrames: true},
+    files: ['content-script.js'],
+  });
+  // results == [{frameId: 0, result: 'success'}, {frameId: 1234, result: null}]
+  for (let output of results) {
+    if (output.result !== 'success') {
+      console.log(`An error occurred in frame ${output.frameId}`);
+    }
+  }
+});
+```
+
+## Wrap up
+
+Manifest version bumps present a rare opportunity to make rethink and modernize extensions APIs. Our
+goal with Manifest V3 is to improve the end user experience by making extensions safer while also
+improving the developer experience. By introducing `chrome.scripting` in Manifest V3, we were able
+to help clean up the Tabs API, to reimagine `executeScript` for a more secure extensions platform,
+and to lay the groundwork for new scripting capabilities that will be coming later this year.
+
+If you have any thoughts or feedback on what we've shared here or Manifest V3 in general, head over
+to the [chromium-extensions](crx-group) Google Group and tell us what you think.
```

Comment:
We link this from other places on the site, and I don't think we want to position it as "feedback channel for any thoughts".  I'd just drop this section.

Reply:
Dropping.